### ISSUE TITLE
Fortran parser: truncate tag line blindly

### ISSUE THREAD

**Issue by QingmingHe** [universal-ctags/ctags#333]
When parsing:

``` Fortran
integer :: i
```

```
i   input.f90   /^in/;" v
```

will be written to tags file. However, what should be in tags file is

```
i   input.f90   /^integer :: i/;"   v
```

This is caused by **truncateTagLine** function in entry.c which uses **strstr** to get the position of tag in current line and truncate text after the tag. In this case, the tag **i** occurs in the first character of **integer** keyword and all the following text will be truncated.
This issue is specific to Fortran parser for that only **truncateLine** is set to be TRUE in fortran.c (I just grep *.c under parser/ directory).
So the easiest way to fix it is to comment out

``` C
        e.truncateLine  = (boolean) (token->tag != TAG_LABEL);
```

in fortran.c. This also make the Fortran parser consistent with the others. However, this fix fails almost all Fortran unit tests in this project and it may not be trivial to update units tags according to this fix. May be a better approach is to teach **truncateTagLine** how to truncate smarter (may not be easy to fix). I can't determine it myself and propose it here to discuss.

**Comment by masatake** [comment]
See also main/get.c.

> % cat /tmp/foo.c
cat /tmp/foo.c
#define d 0
[yamato@x201]~/var/ctags-github% ./ctags -o - /tmp/foo.c
./ctags -o - /tmp/foo.c
d	/tmp/foo.c	/^#de/;"	d	file:

**Comment by masatake** [comment]
strrchr in routines.h can be used to improve the output.
However, I wonder why truncateLine is set TRUE in get.c and fortran.c...
git blame doesn't help us. The field, truncateLine existed when ctags source tree was imported to its subversion repository.

**Comment by QingmingHe** [comment]
strrstr will not improve for some cases. Consider:
```C
int i: // i i i i
```

**Comment by QingmingHe** [comment]
strrstr can't get the perfect output:
```C
#define d 0 // d is here to puzzle you
```
In fact, is there any case where truncateLine is useful? I don't think it make any sense.

**Comment by vhda** [comment]
I don't think that a complete line pattern is necessary. Those entries are Vim Ex commands that are used to search for a pattern within a file and as long as the pattern is unique in the file, then it is sufficient to find the tag.
I've made a quick test using the example Fortran subroutine and added a second procedure called sub2 and this is what I obtained in the tags file:

```
sub	ola.for	/^subr/;"	s
sub2	ola.for	/^subroutine sub2(/;"	s
```

As it can be seen the second subroutine has a different pattern, guaranteeing that each tag can be found correctly. I've also tested this tags file in Vim and it was able to uniquely find each tag as expected.

That said, it is possible that this is an old tags file size optimization feature, but I don't know if such a feature would present any advantage when compared to the use of `--excmd=number`.
